src/plugins/discover/public/application/components/discover_legacy.tsx

```diff
@@ -0,0 +1,328 @@
+  state: AppState;
+  timefilterUpdateHandler: (ranges: { from: number; to: number }) => void;
+  timeRange?: { from: string; to: string };
+  topNavMenu: TopNavMenuData[];
+  updateQuery: (payload: { dateRange: TimeRange; query?: Query }, isUpdate?: boolean) => void;
+  updateSavedQueryId: (savedQueryId?: string) => void;
+  vis?: Vis;
+}
+
+export function DiscoverLegacy({
+  addColumn,
+  fetch,
+  fetchCounter,
+  fetchError,
+  fieldCounts,
+  histogramData,
+  hits,
+  indexPattern,
+  minimumVisibleRows,
+  onAddFilter,
+  onChangeInterval,
+  onMoveColumn,
+  onRemoveColumn,
+  onSkipBottomButtonClick,
+  onSort,
+  opts,
+  resetQuery,
+  resultState,
+  rows,
+  searchSource,
+  setIndexPattern,
+  showSaveQuery,
+  state,
+  timefilterUpdateHandler,
+  timeRange,
+  topNavMenu,
+  updateQuery,
+  updateSavedQueryId,
+  vis,
+}: DiscoverLegacyProps) {
+  const [isSidebarClosed, setIsSidebarClosed] = useState(false);
+  const { TopNavMenu } = getServices().navigation.ui;
+  const { savedSearch, indexPatternList } = opts;
+  const bucketAggConfig = vis?.data?.aggs?.aggs[1];
+  const bucketInterval =
+    bucketAggConfig && search.aggs.isDateHistogramBucketAggConfig(bucketAggConfig)
+      ? bucketAggConfig.buckets?.getInterval()
+      : undefined;
+  const [fixedScrollEl, setFixedScrollEl] = useState<HTMLElement | undefined>();
+
+  useEffect(() => (fixedScrollEl ? opts.fixedScroll(fixedScrollEl) : undefined), [
+    fixedScrollEl,
+    opts,
+  ]);
+  const fixedScrollRef = useCallback(
+    (node: HTMLElement) => {
+      if (node !== null) {
+        setFixedScrollEl(node);
+      }
+    },
+    [setFixedScrollEl]
+  );
+  const sidebarClassName = classNames({
+    closed: isSidebarClosed,
+  });
+
+  const mainSectionClassName = classNames({
+    'col-md-10': !isSidebarClosed,
+    'col-md-12': isSidebarClosed,
+  });
+
+  const sidebarCloseIcon = classNames({
+    'fa-chevron-circle-right': isSidebarClosed,
+    'fa-chevron-circle-left': !isSidebarClosed,
+  });
+
+  return (
+    <I18nProvider>
+      <div className="app-container" data-fetch-counter={fetchCounter}>
+        <h1 className="euiScreenReaderOnly">{savedSearch.title}</h1>
+        <TopNavMenu
+          appName="discover"
+          config={topNavMenu}
+          indexPatterns={[indexPattern]}
+          onQuerySubmit={updateQuery}
+          onSavedQueryIdChange={updateSavedQueryId}
+          query={state.query}
+          setMenuMountPoint={opts.setHeaderActionMenu}
+          savedQueryId={state.savedQuery}
+          screenTitle={savedSearch.title}
+          showDatePicker={indexPattern.isTimeBased()}
+          showSaveQuery={showSaveQuery}
+          showSearchBar={true}
+          useDefaultBehaviors={true}
+        />
+        <main className="container-fluid">
+          <div className="row">
+            <div
+              className={`col-md-2 dscSidebar__container collapsible-sidebar ${sidebarClassName}`}
+              id="discover-sidebar"
```

Comment:
Technically we should be getting rid of Bootstrap classes here, but there can be a follow up to replace this with EuiFlexGroup

Reply:
Thankfully most of these happen in the larger grid PR.

Reply:
@cchaos @snide yes, most of this was done in the data grid PR. focus of this PR was to unblock other deangularization and nearing the GAP between those 2 PRs. thx for you synchronous review feedback. happy to remove more old bits to sunset the font that used to be awesome